tests/tests_geomstats/test_discrete_curves.py

```diff
@@ -1,315 +1,617 @@
+        n_samples_list = random.sample(range(2, 5), 2)
+        n_points_list = random.sample(range(2, 5), 2)
+        n_vecs_list = random.sample(range(2, 5), 2)
+
+        def random_point_belongs_test_data(self):
+            smoke_space_args_list = [(self.s2,), (self.r2,)]
+            smoke_n_points_list = [1, 2]
+            return self._random_point_belongs_test_data(
+                smoke_space_args_list,
+                smoke_n_points_list,
+                self.space_args_list,
+                self.n_points_list,
+            )
 
-        self.assertAllClose(result, expected)
+        def projection_belongs_test_data(self):
+            return self._projection_belongs_test_data(
+                self.space_args_list, self.shape_list, self.n_points_list
+            )
 
-    def test_l2_metric_log_and_exp(self):
-        """Test that exp and log are inverse maps."""
-        tangent_vec = self.l2_metric_s2.log(point=self.curve_b, base_point=self.curve_a)
-        result = self.l2_metric_s2.exp(tangent_vec=tangent_vec, base_point=self.curve_a)
-        expected = self.curve_b
+        def to_tangent_is_tangent_test_data(self):
+            return self._to_tangent_is_tangent_test_data(
+                DiscreteCurves,
+                self.space_args_list,
+                self.shape_list,
+                self.n_vecs_list,
+            )
 
-        self.assertAllClose(result, expected)
+        def random_tangent_vec_is_tangent_test_data(self):
+            return self._random_tangent_vec_is_tangent_test_data(
+                DiscreteCurves, self.space_args_list, self.n_vecs_list
+            )
 
-    def test_l2_metric_inner_product_vectorization(self):
-        """Test the vectorization inner_product."""
-        n_samples = self.n_discretized_curves
-        curves_ab = self.l2_metric_s2.geodesic(self.curve_a, self.curve_b)
-        curves_bc = self.l2_metric_s2.geodesic(self.curve_b, self.curve_c)
-        curves_ab = curves_ab(self.times)
-        curves_bc = curves_bc(self.times)
+    testing_data = DiscreteCurvesTestData()
 
-        tangent_vecs = self.l2_metric_s2.log(point=curves_bc, base_point=curves_ab)
 
-        result = self.l2_metric_s2.inner_product(tangent_vecs, tangent_vecs, curves_ab)
+class TestL2CurvesMetric(RiemannianMetricTestCase, metaclass=Parametrizer):
+    metric = connection = L2CurvesMetric
+    skip_test_exp_belongs = True
+    skip_test_exp_shape = True
+    skip_test_log_shape = True
+    skip_test_exp_geodesic_ivp = True
+    skip_test_parallel_transport_ivp_is_isometry = True
+    skip_test_parallel_transport_bvp_is_isometry = True
+    skip_test_dist_is_norm_of_log = tf_backend()
+    skip_test_dist_is_symmetric = tf_backend()
+    skip_test_squared_dist_is_symmetric = tf_backend()
+    skip_test_inner_product_is_symmetric = tf_backend()
 
-        self.assertAllClose(gs.shape(result), (n_samples,))
+    class L2CurvesMetricTestData(_RiemannianMetricTestData):
```

Comment:
As per @lpereira95 refactoring, we do not have nested testdata classes anymore. This should go in: https://github.com/geomstats/geomstats/tree/master/tests/data